okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java

```diff
@@ -151,6 +156,20 @@ public void onResponse(Call _call, Response response) {
         }
     }
 
+    private void closeResponseBody(SettableFuture<Response> future) {
+        try {
+            Response response = future.get();
+            if (response.body() != null) {
```

Comment:
I think to make intent clearer, you want something along the lines of:

```java
if (future.isDone()) {
  Response response = Futures.getDone(future);
  if (response.body() != null) {
    response.close();
  } 
}
```

I think you may have to deal with the `ExecutionException`, or just use `Futures.getUnchecked`.

Reply:
I would actually move the future setting code in here, and call the method `handleInterruption`

Also docs as to why what you're doing works, something along these lines but obviously better.

```java
private InterruptedIOException handleInterruption(SettableFuture<Response> future) {
  getDelegate().cancel();
  // (1) if the success callback gets called before we try to set the exception, we will then close the response
  // since it won't get processed

  // (2) if the success callback gets called after we try to set the exception, the 
  // callback will close the response
  if (!future.setException(new InterruptedIOException("Call cancelled via interruption")) {
    // Future is already set via the callback
    try {
      Response response = Futures.getDone(future);
      // Future set via the callback but successfully, need to clean up.
      if (response.body() != null) {
        response.close();
      }
    } catch (ExecutionException e) {
      // Future set via the callback but exceptionally.
      // Empty clause which will be dealt with via the fall through block
    }
  }

  Thread.currentThread().interrupt();
  return new InterruptedIOException("Call cancelled via interruption");
}
```